sound/hda/hdac_stream.c

```diff
@@ -405,11 +406,20 @@ int snd_hdac_stream_setup_periods(struct hdac_stream *azx_dev)
 {
 	struct hdac_bus *bus = azx_dev->bus;
 	struct snd_pcm_substream *substream = azx_dev->substream;
-	struct snd_pcm_runtime *runtime = substream->runtime;
+	struct snd_compr_stream *cstream = azx_dev->cstream;
+	struct snd_pcm_runtime *runtime = NULL;
+	struct snd_dma_buffer *dma_buffer_p;
 	__le32 *bdl;
 	int i, ofs, periods, period_bytes;
 	int pos_adj, pos_align;
 
+	if (substream) {
+		runtime = substream->runtime;
+		dma_buffer_p = snd_pcm_get_dma_buf(substream);
+	} else {
+		dma_buffer_p = snd_pcm_get_dma_buf(cstream);
```

Comment:
the _p suffix can be confusing, it could be taken as 'playback'. Do we need it or could we just use "dma_buffer"?

Reply:
Fixed in v6. Renamed to common name for struct snd_dma_buffer local: dmab.